raco/backends/myria/myria.py

```diff
@@ -687,6 +717,10 @@ class MyriaSplitConsumer(algebra.UnaryOperator, MyriaOperator):
     """A Myria SplitConsumer"""
 
     def __init__(self, input):
+        assert isinstance(input, MyriaSplitProducer)
+        if not hasattr(input, 'consumers'):
```

Comment:
This is for the rule `RemoveSingleSplit`, which removes a `SplitProducer` that only has one `SplitConsumer` followed by a `ShuffleProducer`. So the `SplitProducer` needs to know its consumers.